src/reason_pprint_ast.ml

```diff
@@ -6301,27 +6301,14 @@ class printer  ()= object(self:'self)
     SourceMap (e.pexp_loc, param)
 
   method label_x_expression_params xs =
-    let is_direct x = x.pexp_attributes = [] && match x.pexp_desc with
-      | Pexp_array _
-      | Pexp_record _
-      | Pexp_object {pcstr_fields = _}
-      | Pexp_construct ( {txt= Lident"()"}, None) -> true
-      | _ -> false
-    in
-    let head, xs = match xs with
-      | (Nolabel, x) :: xs when is_direct x ->
-        ([self#simplifyUnparseExpr x], xs)
-      | _ -> ([], xs)
-    in
-    let tail, xs = match List.rev xs with
-      | (Nolabel, x) :: xs when is_direct x ->
-        ([self#simplifyUnparseExpr x], List.rev xs)
-      | _ -> ([], xs)
-    in
-    let xs = if xs = [] then [] else
-        [makeTup (List.map self#label_x_expression_param xs)]
+    let xs = (match xs with
+      (* function applications with unit as only argument should be printed differently
+       * e.g. print_newline(()) should be printed as print_newline() *)
+      | [(_, ({pexp_attributes = []; pexp_desc = Pexp_construct ( {txt= Lident "()"}, None)} as x))]
```

Comment:
You should restrict the rewriting to the case where the label is `Nolabel`.

Reply:
Indeed, thx!